### ISSUE TITLE
[Autocomplete Item] Context is not provided to assistive technology when in focus

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#12505]
**monday.com sync:** #18005841588

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

When the `autocomplete-item` component is in focus it does not provide context to assistive technologies.

Discovered during the verification of adding an `aria-label` to the parent `autocomplete` when results are displayed when interacting with the component: https://github.com/Esri/calcite-design-system/issues/12433#issuecomment-3064436742

### Actual Behavior

When an `autocomplete-item` is in focus the component does not provide context to assistive technologies.

### Expected Behavior

When an `autocomplete-item` is in focus the `heading`, or if provided `label` is communicated to assistive technologies.

For instance:

```html
<!-- Heading communicated when in focus -->
...
<calcite-autocomplete-item value="elderberry" heading="Elderberry"></calcite-autocomplete-item>
...

<!-- Label communicated when in focus -->
...
<calcite-autocomplete-item value="a-value" heading="Displayed value" label="A more descriptive label"></calcite-autocomplete-item>
...
```

### Reproduction Sample

https://codepen.io/isaacbraun/pen/pvjvbvX

### Reproduction Steps

1. Activate NVDA or JAWS
2. Open the [sample](https://codepen.io/isaacbraun/pen/pvjvbvX)
3. Search for a place like `"San Diego"`
4. Arrow down into the first `autocomplete-item` result
5. Observe no context is provided for the item's `heading` or `label`

### Reproduction Version

`3.3.0-next.33`

### Working W3C Example/Tutorial

https://www.w3.org/WAI/WCAG22/Understanding/on-input.html

### Relevant Info

Violates WCAG SC [3.2.2: On Input](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html) (Level A)

Blocks: #12515 

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by annelfitz** [comment]
This was just brought up to us by the Experience Builder team, noticed within our Search component. 
<img width="1888" height="780" alt="Image" src="https://github.com/user-attachments/assets/81607ce8-b7b4-407f-a987-ec0d7f711624" />
See https://devtopia.esri.com/WebGIS/arcgis-web-components/issues/11412 for more details.

**Comment by geospatialem** [comment]
Team members have a prototype in progress and are very close to a resolution, however since we are very close to release and want to ensure proper testing and mitigate risk, we are moving the above effort into the next release where we are targeting adoption early in the cycle for more testing and mitigation of risk across implementations and scenarios of the component.

cc @Esri/calcite-pes @jcfranco @Amretasre002762670

**Comment by Amretasre002762670** [comment]
Moving this to `2- ready for Dev` since it is not currently being actively worked on.
